src/sql/base/browser/ui/infoButton/infoButton.ts

```diff
@@ -0,0 +1,183 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Button as sqlButton } from 'sql/base/browser/ui/button/button';
+import * as DOM from 'vs/base/browser/dom';
+import { IButtonOptions } from 'vs/base/browser/ui/button/button';
+import { URI } from 'vs/base/common/uri';
+
+type IUserFriendlyIcon = string | URI | { light: string | URI; dark: string | URI };
+
+export interface IInfoButtonOptions extends IButtonOptions {
+	description: string,
+	buttonMaxHeight: number,
+	iconClass: string,
+	iconHeight: number,
+	iconWidth: number,
+	iconPath: IUserFriendlyIcon,
+	textTitle: string,
+	buttonMaxWidth: number,
+}
+
+export class InfoButton extends sqlButton {
+	private _container: HTMLElement;
+	private _main: HTMLElement;
+	private _iconContainer: HTMLElement;
+	private _iconElement: HTMLElement;
+	private _textContainer: HTMLElement;
+	private _pTitle: HTMLElement;
+	private _pDesc: HTMLElement;
+
+	private _description: string = '';
+	private _buttonMaxHeight: number = 0;
+	private _iconClass: string = '';
+	private _iconHeight: number = 0;
+	private _iconWidth: number = 0;
+	private _iconPath: IUserFriendlyIcon = { light: '', dark: '' };
+	private _textTitle: string = '';
+	private _buttonMaxWidth: number = 0;
+
+	constructor(container: HTMLElement, options?: IInfoButtonOptions) {
+		super(container, options);
+		this._container = container;
+
+		{ // Creates the elements
+			this._container.style.display = 'flex';
+			this._container.style.justifyContent = 'space-around';
+
+			this._main = document.createElement('div');
+			DOM.addClass(this._main, 'flexContainer');
+			this._main.style.cursor = 'pointer';
+			this._main.style.backgroundColor = '#FFFFFF';
```

Comment:
this is not theme-aware, the background will always be white. there are ways to do this in the right way, but I'd remove the background color and use the default values. If you really want to control the background color, please refer to the style() method in the base class, and also the new color needs to be added to the button.component.ts in the modelComponents folder's attachButtonStyler calls

Reply:
I added a stylesheet specific to infoButton to handle button background color, box-shadow and hover box-shadow change. These are candidates for a theme-aware solution.